data/mods/ssb/moves.ts

```diff
@@ -968,6 +968,34 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		type: "Rock",
 	},
 
+	// smely socks
+	stockholmsyndrome: {
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		shortDesc: "Curses and traps foe. User loses 1/2 HP.",
+		name: "Stockholm Syndrome",
+		pp: 5,
+		priority: 0,
+		flags: {bypasssub: 1},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Curse', target);
+			this.add('-anim', source, 'Block', target);
+		},
+		onHit(target, source, move) {
+			if (!target.addVolatile('curse')) return false;
+			this.directDamage(source.maxhp / 2, source, source);
+			target.addVolatile('trapped', source, move, 'trapper');
+		},
```

Comment:
```suggestion
		onHit(target, source, move) {
			let success = false;
			if (!target.volatiles['curse']) {
				this.directDamage(source.maxhp / 2, source, source);
				target.addVolatile('curse');
				success = true;
			}
			return target.addVolatile('trapped', source, move, 'trapper') || success;
		},
```

Reply:
better order of events + doesn't auto fail if the mon can still trap but not inflict curse